backend/lib/auth.js

```diff
@@ -99,20 +99,26 @@ function init(app) {
 }
 
 function login(credentials) {
-    db.findOne('users', { email: credentials.email }).then(doc => {
-        if (doc) {
-            return bcrypt.compare(credentials.passphrase, doc.passphrase, (err, res) => {
-                if (res) {
-                    // Passwords match
-                    return doc
-                }
-                // Passwords don't match
-                return 'Passwords do not match'
-            })
-        }
-    }).catch(err => {
-        console.log('error in login', err)
-        return err
+    return new Promise( (resolve, reject) => {

```

Comment:
We don't need to create a new Promise here.  

I'd also usually don't like mixing together Promises and callbacks within the same app - it can cause confusion sometimes.  Potentially try something like this?
```js
function login(credentials) {
    return db.findOne('users', { email: credentials.email }).then(doc => {
        return bcrypt.compare(credentials.passphrase, doc[0].passphrase)
    }).catch(err => {
        console.log('error in login ---->', err)
        return err
    })
}
```

Reply:
+1 had to refactor a little bit too